src/StoreKeeper/WooCommerce/B2C/Backoffice/Helpers/OverlayRenderer.php

```diff
@@ -79,6 +91,66 @@ public function renderError($message, $description = '')
         $this->renderContentItem($content);
     }
 
+    public function renderMemoryExhaustError($message, $description = '')
+    {
+        $message = esc_html($message);
+        $description = esc_html($description);
+
+        $firstExplanationText = __('Your allowed memory size has been exhausted. Here are some solution that may solve the issue.', I18N::DOMAIN);
+        $secondExplanationText = __('But before trying some of these you can check what is the defined memory_limit of your site. You can check this by Clicking Tools > Site Health > Info > Server. Here you can check the current settings of your hosting.', I18N::DOMAIN);
+
+        $serverInfoImagePath = plugin_dir_url(__FILE__).'../static/server-information.png';
+
+        $instructionHeading = __('After identifying the value defined on memory and still the error persist.', I18N::DOMAIN);
+        $firstInstruction = sprintf(
+            __('You try can increasing the memory by adding %s on the wp-config.', I18N::DOMAIN),
+            "<code>define('WP_MEMORY_LIMIT', '256M');</code>"
+        );
+        $firstInstructionFirstBullet = __('256M is a size sample you can put and then do a full sync.', I18N::DOMAIN);
+
+        $secondInstruction = __('You can try executing this command via ssh.', I18N::DOMAIN);
+        $secondInstructionFirstBullet = __('Go to the public_html folder of your webshop (e.g. /webshop/public_html/)', I18N::DOMAIN);
+        $secondInstructionSecondBullet = sprintf(
+            __('Execute %s.', I18N::DOMAIN),
+            '<code>wp sk sync-woocommerce-full-sync</code>'
+        );
+
+        $thirdInstruction = __('Contact your hosting provider.', I18N::DOMAIN);
+        $thirdInstructionFirstBullet = __('If you are not comfortable in trying the methods above, or it did not work for you. You can talk to your hosting provider about having them increase your memory limit.', I18N::DOMAIN);
+
+        $content = <<<HTML
```

Comment:
I see a lot of unescaped variables in here

Reply:
Added escaping